### PR TITLE
DCMAW-10243: Add pretty DNS for sessions api

### PR DESCRIPTION
**Pull request by sandymay-dd**
​DCMAW-10243

### What changed
Adds pretty DNS for sessions api
Rename public api to sessions api

### Why did it change
Programme standard
Tied the API name with the functionality it servers

### Testing

Tests run locally against deployed stack:
image

image

## Checklists

- [ ] There is a ticket raised for this PR that is present in the branch name
- [ ] No PII data logged. [See guidance here](https://govukverify.atlassian.net/wiki/spaces/DCMAW/pages/3502407722/PII+Logging+Considerations)
- [ ] Demo to a BA, TA, and the team.
- [ ] Update [README](./blob/main/README.md) with any new instructions or tasks

### CODE REVIEW

**Review by jmooney-dd — CHANGES_REQUESTED** [review]
Please can we update the references in the README

**Comment by jmooney-dd** [comment]
> Please can we update the references in the README

Ignore me. I can't dismiss this tho, can you @sandymay-dd

**Review by ahickmann-gds — APPROVED** [review]

File: backend-api/generate_env_file.sh [hunk 1]
```diff
@@ -11,8 +11,8 @@ fi
 
 echo "Generating .env file for the $SAM_STACK stack"
 
-PUBLIC_API_URL=$(aws cloudformation describe-stacks --stack-name "$SAM_STACK" --query "Stacks[0].Outputs[?OutputKey=='PublicApiUrl'].OutputValue" --output text)
+SESSIONS_API_URL=$(aws cloudformation describe-stacks --stack-name "$SAM_STACK" --query "Stacks[0].Outputs[?OutputKey=='PublicApiUrl'].OutputValue" --output text)
```
Comment (ahickmann-gds) [inline backend-api/generate_env_file.sh:14]: Does this `PublicApiUrl` need to be `SessionsApiUrl`?

Comment (sandymay-dd) [reply backend-api/generate_env_file.sh:14]: nice spot - I missed this file in testing

**Comment by sandymay-dd** [comment]
> Please can we update the references in the README, I think it's just the reference to Regional API?

it still is a Regional API (that's the technical AWS term for the type of API)

File: backend-api/template.yaml [hunk 2]
```diff
@@ -1113,9 +1156,14 @@ Resources:
         - !Ref AWS::NoValue
 
 Outputs:
-  PublicApiUrl:
-    Description: Public API Gateway base URL
-    Value: !Sub https://${BackendAsyncCredentialPublicApi}.execute-api.${AWS::Region}.amazonaws.com/${Environment}/
+  SessionsApiUrl:
```
Comment (ahickmann-gds) [inline backend-api/template.yaml:1159]: Out of curiosity, would this work instead? 🤔 

`Value: !Sub https://${SessionsApiDomainName}`

**Review by ahickmann-gds — APPROVED** [review]

**Review by jmooney-dd — APPROVED** [review]

File: backend-api/template.yaml (hunk 2 above)
Comment (sandymay-dd) [reply backend-api/template.yaml:1159]: It should do yes - I'll raise a follow-up PR

Comment (ahickmann-gds) [reply backend-api/template.yaml:1159]: I just thought it might be nicer but it doesn't have to change

**Pull request merged**